packages/client/hooks/useHover.ts

```diff
@@ -0,0 +1,29 @@
+import {useRef, useCallback, useState} from 'react'
+
+const useHover = <T extends HTMLElement>(): [(node?: T | null) => void, boolean] => {
```

Comment:
The main pros I think are that having a reusable hook (a flexible abstraction) helps keep code DRY, readability clear, and implementation simple.

The con is that it's using refs to achieve this flexibility. Using a virtual Dom pattern like you described means that we'd no longer need to use the ref (which can be a bit of an anti-pattern in React). I felt this was a small enough component that wouldn't be re-rendering it's children as much and thought the reusability outweighed the potential performance considerations. 

I do like your use of `mouseenter` and `mouseleave` in your example since it won't trigger on interactions with children like `mouseover` and `mouseout` will. I've updated useHover to use those now.